src/main/java/com/blackduck/integration/stepworkflow/jenkins/JenkinsStepWorkflow.java

```diff
@@ -1,39 +1,59 @@
+import com.blackduck.integration.rest.client.IntHttpClient;
+import com.blackduck.integration.rest.response.Response;
+import com.blackduck.integration.stepworkflow.StepWorkflow;
+import com.blackduck.integration.stepworkflow.StepWorkflowResponse;
+import com.google.gson.Gson;
+import com.blackduck.integration.jenkins.extensions.JenkinsIntLogger;
+import com.blackduck.integration.phonehome.PhoneHomeClient;
+import com.blackduck.integration.phonehome.PhoneHomeResponse;
+import com.blackduck.integration.phonehome.PhoneHomeService;
+import com.blackduck.integration.phonehome.google.analytics.GoogleAnalyticsConstants;
+import com.blackduck.integration.phonehome.request.PhoneHomeRequestBody;
+import com.blackduck.integration.phonehome.request.PhoneHomeRequestBodyBuilder;
+import com.blackduck.integration.rest.proxy.ProxyInfo;
+import com.google.gson.JsonSyntaxException;
+import hudson.AbortException;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
 
+import java.io.IOException;
 import java.util.Optional;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
-import org.apache.http.impl.client.HttpClientBuilder;
-
-import com.google.gson.Gson;
-import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
-import com.synopsys.integration.jenkins.wrapper.JenkinsVersionHelper;
-import com.synopsys.integration.phonehome.PhoneHomeClient;
-import com.synopsys.integration.phonehome.PhoneHomeResponse;
-import com.synopsys.integration.phonehome.PhoneHomeService;
-import com.synopsys.integration.phonehome.google.analytics.GoogleAnalyticsConstants;
-import com.synopsys.integration.phonehome.request.PhoneHomeRequestBody;
-import com.synopsys.integration.phonehome.request.PhoneHomeRequestBodyBuilder;
-import com.synopsys.integration.stepworkflow.StepWorkflow;
-import com.synopsys.integration.stepworkflow.StepWorkflowResponse;
-
-import hudson.AbortException;
-
 public abstract class JenkinsStepWorkflow<T> {
     protected final JenkinsIntLogger logger;
     protected final JenkinsVersionHelper jenkinsVersionHelper;
+    private final static int TIMEOUT_IN_SECONDS = 20;
+    private final Gson gson;
+    private final IntHttpClient intHttpClient;
+    private final static String CREDENTIALS_PATH = "https://static-content.app.blackduck.com/detect/analytics/creds.json";
 
     public JenkinsStepWorkflow(JenkinsIntLogger logger, JenkinsVersionHelper jenkinsVersionHelper) {
         this.logger = logger;
         this.jenkinsVersionHelper = jenkinsVersionHelper;
+        this.gson = new Gson();
+        this.intHttpClient = new IntHttpClient(
+                new SilentIntLogger(),
+                gson,
+                TIMEOUT_IN_SECONDS,
+                true,
+                ProxyInfo.NO_PROXY_INFO
+        );
     }
 
     protected abstract PhoneHomeRequestBodyBuilder createPhoneHomeBuilder();
```

Comment:
I should probably know this, but could you share where this is used/implemented?

Reply:
You mean the library? It is implemented in phone home client, also the code is the same as in Detect.

Reply:
I noticed that this `createPhoneHomeBuilder()` method is abstract and got curious where the non-abstract version(s) of it is/are. I don't see such a method in this repo (except for a test) nor in phone home client and Detect.